### PR TITLE
Improve logging message for retryable background IO errors

### LINKED ISSUE #2312: Improve error handling and logging for Retryable IO Errors
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

In [server.cc](https://github.com/apache/kvrocks/blob/unstable/src/server/server.cc) file, the code checks every minute if the storage is in a retryable IO error state and resumes the database if necessary.
```c
 if (counter != 0 && counter % 600 == 0 && storage->IsDBInRetryableIOError()) {
    storage->GetDB()->Resume();
    LOG(INFO) << "[server] Schedule to resume DB after retryable IO error";
    storage->SetDBInRetryableIOError(false);
  }
```
Concerns:
1. The code resumes the database and sets the retryable IO error state to false without checking if the resume operation itself succeeded. 
2. logs INFO message which may be ERROR or WARNING instead

### Solution

example code to resolve this:
```c
if (counter != 0 && counter % 600 == 0 && storage->IsDBInRetryableIOError()) {
  auto status = storage->GetDB()->Resume();
  if (status.ok()) {
    LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error: " << status.ToString();
    storage->SetDBInRetryableIOError(false);
  } else {
    LOG(ERROR) << "[server] Failed to resume DB after retryable IO error: " << status.ToString();
    // Additional error handling, such as retrying or notifying the administrator
  }
}
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by VasuDevrani**
This closes #2312

### CODE REVIEW

**Review by git-hulk — DISMISSED** [review]

File: src/server/server.cc [hunk 1]
```diff
@@ -808,9 +808,13 @@ void Server::cron() {
     // In order to properly handle all possible situations on rocksdb, we manually resume here
     // when encountering no space error and disk quota exceeded error.
     if (counter != 0 && counter % 600 == 0 && storage->IsDBInRetryableIOError()) {
-      storage->GetDB()->Resume();
-      LOG(INFO) << "[server] Schedule to resume DB after retryable IO error";
-      storage->SetDBInRetryableIOError(false);
+      rocksdb::Status status = storage->GetDB()->Resume();
+      if (status.ok()) {
+        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error: " << status.ToString();
```
Comment (PragmaTwice) [inline src/server/server.cc]:
```suggestion
        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error.";
```

Comment (PragmaTwice) [reply src/server/server.cc]: It's meaningless to log an `OK` status.

**Review by mapleFU — DISMISSED** [review]

File: src/server/server.cc [hunk 2]
```diff
@@ -808,9 +808,13 @@ void Server::cron() {
     // In order to properly handle all possible situations on rocksdb, we manually resume here
     // when encountering no space error and disk quota exceeded error.
     if (counter != 0 && counter % 600 == 0 && storage->IsDBInRetryableIOError()) {
-      storage->GetDB()->Resume();
-      LOG(INFO) << "[server] Schedule to resume DB after retryable IO error";
-      storage->SetDBInRetryableIOError(false);
+      rocksdb::Status status = storage->GetDB()->Resume();
```
Comment (caipengbo) [inline src/server/server.cc]:
```suggestion
      auto s = storage->GetDB()->Resume();
```

File: src/server/server.cc [hunk 3]
```diff
@@ -808,9 +808,13 @@ void Server::cron() {
...
+      if (status.ok()) {
+        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error.";
```
Comment (caipengbo) [inline src/server/server.cc]:
```suggestion
        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error";
```

**Review by caipengbo — DISMISSED** [review]

**Review by PragmaTwice — DISMISSED** [review]

**Review by PragmaTwice — DISMISSED** [review]

File: src/server/server.cc [hunk 4]
```diff
@@ -808,9 +808,13 @@ void Server::cron() {
     // In order to properly handle all possible situations on rocksdb, we manually resume here
     // when encountering no space error and disk quota exceeded error.
     if (counter != 0 && counter % 600 == 0 && storage->IsDBInRetryableIOError()) {
-      storage->GetDB()->Resume();
-      LOG(INFO) << "[server] Schedule to resume DB after retryable IO error";
-      storage->SetDBInRetryableIOError(false);
+      auto s = storage->GetDB()->Resume();
+      if (status.ok()) {
```
Comment (caipengbo) [inline src/server/server.cc]:
```suggestion
      if (s.ok()) {
```

File: src/server/server.cc [hunk 5]
```diff
@@ -808,9 +808,13 @@ void Server::cron() {
...
+        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error";
+        storage->SetDBInRetryableIOError(false);
+      } else {
+        LOG(ERROR) << "[server] Failed to resume DB after retryable IO error: " << status.ToString();
```
Comment (caipengbo) [inline src/server/server.cc]:
```suggestion
        LOG(ERROR) << "[server] Failed to resume DB after retryable IO error: " << s.ToString();
```

**Review by caipengbo — DISMISSED** [review]

File: src/server/server.cc [hunk 6]
```diff
@@ -808,9 +808,13 @@ void Server::cron() {
...
-      storage->GetDB()->Resume();
-      LOG(INFO) << "[server] Schedule to resume DB after retryable IO error";
-      storage->SetDBInRetryableIOError(false);
+      auto s = storage->GetDB()->Resume();
+      if (s.ok()) {
+        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error";
+        storage->SetDBInRetryableIOError(false);
+      } else {
+        LOG(ERROR) << "[server] Failed to resume DB after retryable IO error: " << s.ToString();
+      }
```
Comment (PragmaTwice) [inline src/server/server.cc]:
```suggestion
      if (s.ok()) {
        LOG(WARNING) << "[server] Successfully resumed DB after retryable IO error";
      } else {
        LOG(ERROR) << "[server] Failed to resume DB after retryable IO error: " << s.ToString();
      }
      storage->SetDBInRetryableIOError(false);
```

Comment (PragmaTwice) [reply src/server/server.cc]: We should keep the original behavior...

**Review by PragmaTwice — CHANGES_REQUESTED** [review]
See above.

**Review by PragmaTwice — APPROVED** [review]

**Pull request merged**